### PR TITLE
OpenCV 4.8.0

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
**Description:**
- Updated OpenCV to 4.8.0 version
- Added OpenVINO dependency to optimize DNN module to faster perform inference of DL models
- Updated macOS SDK version for x86_64 because of [this patch](https://github.com/opencv/opencv/pull/23394) in 4.8.0
- Pinned eigen dependency, because newer patch version (released today) has issues with compilation. 

**Checklist**
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [X] Reset the build number to `0` (if the version changed)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -31,16 +31,18 @@ package:
...
-    sha256: 8df0079cdbe179748a18d44731af62a245a45ebf5085223dc03133954c662973
+    sha256: cbf47ecc336d2bff36b0dcd7d6c179a9bb59e805136af6b9670ca944aef889bd
     patches:
       # backport https://github.com/opencv/opencv/pull/21611 (unmerged as of 06/2023)
       - patches_opencv/0001-Add-installation-of-pip-metadata-from-cmake.patch
       - patches_opencv/0002-delete-lines-that-download-opencv.patch
       - patches_opencv/0003-find-pkgconfig-on-windows.patch
       - patches_opencv/0004-fix-detection-for-protobuf-23.x.patch
+      # reverted https://github.com/opencv/opencv/pull/23394 since does not work for macOS 10.9 x86_64
+      - patches_opencv/0005-revert-cocoa-wheel-scroll.patch  # [osx and x86_64]
```
Comment (hmaarrfk) [inline recipe/meta.yaml:42]:
```suggestion
      # - patches_opencv/0005-revert-cocoa-wheel-scroll.patch  # [osx and x86_64]
```

with your increase in the SDK this should be fine right? Thanks for looking into this, I too agree that the default SDK is likely too low, but it has been difficult to change it.

Comment (ilya-lavrenov) [reply recipe/meta.yaml]: yes, PR is ready now.

**Comment by ilya-lavrenov** [comment]
> Thanks for the help on this front!
> 
> > Added OpenVINO dependency to optimize DNN module to faster perform inference of DL models
> 
> Are you comfortable with conda-forge? do you maintain any other conda-forge feedstocks??
> 
> To debug the build failures, it may be really helpful to disable much of the python build matrix to avoid unecessary CI usage

Hi!
Yes, I am an author of:
- https://github.com/conda-forge/openvino-feedstock
- https://github.com/conda-forge/staged-recipes/pull/23102

and maintainer of:
- https://github.com/conda-forge/pugixml-feedstock
- https://github.com/conda-forge/xbyak-feedstock

File: recipe/meta.yaml [hunk 2]
```diff
@@ -85,7 +85,7 @@ requirements:
   host:
     - python
     - numpy
-    - eigen
+    - eigen =3.4.0
```
Comment (hmaarrfk) [inline recipe/meta.yaml:88]: Why this change

Comment (ilya-lavrenov) [reply recipe/meta.yaml:88]: As it's stated in the description: `Pinned eigen dependency, because newer patch version (released today) has issues with compilation.`

**Review by hmaarrfk — APPROVED** [review]
Great work. Mostly I want to give others a chance to chime in on adding you as a maintainer

File: recipe/meta.yaml [hunk 3]
```diff
@@ -280,3 +282,4 @@ extra:
     - hajapy
     - ocefpaf
     - hmaarrfk
+    - ilya-lavrenov
```
Comment (ilya-lavrenov) [inline recipe/meta.yaml:285]:
```suggestion
```

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

File: recipe/meta.yaml (hunk 2 above)
Comment (traversaro) [reply recipe/meta.yaml:88]: Just FYI, that version of eigen was not actually meant to be released: https://github.com/conda-forge/eigen-feedstock/issues/39 .
